From a47bccd51099b5775d99b070d798e657d7e05995 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Tue, 14 Jul 2020 10:01:38 +0700
Subject: [PATCH 253/433] ARM: dts: r8a7743: Add 3DX (PowerVR SGX544MP2) node

Add definition of 3DX graphics module to the R8A7743 (RZ/G1M) SoC dtsi.

Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
Signed-off-by: Duy Bao Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7743.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index aed0674..bda260e 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -1770,6 +1770,17 @@
 			renesas,#wpf = <1>;
 		};
 
+		sgx@fd800000 {
+			compatible = "renesas,sgx";
+			reg = <0 0xfd800000 0 0x10000>;
+			interrupts = <0 119 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "irq_sgx";
+			clocks = <&mstp1_clks R8A7743_CLK_3DG>,
+				 <&mstp8_clks R8A7743_CLK_IPMMU_SGX>;
+			clock-names = "sgx_clk", "ipmmu_clk";
+			power-domains = <&sysc R8A7743_PD_SGX>;
+		};
+
 		pci0: pci@ee090000 {
 			compatible = "renesas,pci-r8a7743",
 				     "renesas,pci-rcar-gen2";
-- 
2.7.4

